app/controllers/podcasts_controller.rb

```diff
@@ -58,30 +60,28 @@ def skip_bullet
     Bullet.enable = previous_value
   end
 
-  def valid_image?
-    image = params.dig("podcast", "image")
-    return true unless image
+  def valid_images?
+    images = podcast_params.select { |k, _v| IMAGE_KEYS.include?(k) }
+    return true if images.blank?
 
-    @podcast = Podcast.new(podcast_params.except(:image))
+    # Create the podcast object to add errors to for the view
+    @podcast = Podcast.new(podcast_params.except(*IMAGE_KEYS))
     @podcast.creator = current_user
-    return true if valid_image_file?(image) && valid_filename?(image)
+    return true if valid_image_files_and_names?(images)
 
     @podcasts = Podcast.available.order(title: :asc)
     @podcast_index = true
     false
   end
 
-  def valid_image_file?(image)
-    return true if file?(image)
-
-    @podcast.errors.add(:image, IS_NOT_FILE_MESSAGE)
-    false
-  end
+  def valid_image_files_and_names?(images)
+    images.each do |field, image|
+      @podcast.errors.add(field, IS_NOT_FILE_MESSAGE) unless file?(image)
+      break if @podcast.errors.any?
```

Comment:
I'm breaking here to avoid doing extra logic once the first error is found. These types of errors are _very_ rare so it's not necessary to always perform all the checks at once.

Reply:
This could mean some relevant error messages are not given back to the user but I dont think that is a huge deal.